test/correctness/fuzz_bounds.cpp

```diff
@@ -0,0 +1,512 @@
+#include "Halide.h"
+#include <array>
+#include <random>
+#include <stdio.h>
+#include <time.h>
+
+namespace {
+
+using std::map;
+using std::string;
+using namespace Halide;
+using namespace Halide::Internal;
+
+#define internal_assert _halide_user_assert
+
+const int fuzz_var_count = 5;
+
+// use std::mt19937 instead of rand() to ensure consistent behavior on all systems
```

Comment:
I don't think we need to justify this choice. rand is basically deprecated anyway